tests/test_fact_manager.py

```diff
@@ -0,0 +1,439 @@
+
+    # ..and verify it is deleted.
+    assert not await fm.exists('test127', 'en')
+
+
+@pytest.mark.asyncio
+async def test_fm_delete_no_mfd_flag(test_fm_fx):
+    """
+    Verify deleting a fact NOT marked mfd fails, throwing a psycopg2.ProgrammingError
+    """
+    fm = test_fm_fx
+
+    with pytest.raises(psycopg2.ProgrammingError):
+        await fm.delete('prep', 'en')
+
+
+@pytest.mark.asyncio
+async def test_mfdlist_returns_list(test_fm_fx):
+    """
+    mfd_list method must return a list.
+    """
+    fm = test_fm_fx
+
+    assert isinstance(await fm.mfd_list(), list)
+
+
+@pytest.mark.asyncio
+async def test_fm_destroy(test_fm_fx):
+    """
+    Verify _destroy function works properly.
+    """
+    fm = test_fm_fx
+
+    await fm._destroy('test', 'ru')
+
+
+@pytest.mark.asyncio
+async def test_edit_message_bad_fact(test_fm_fx):
+    """
+    Verify attempt to edit a fact that does not exist throws exception.
+    """
+    fm = test_fm_fx
+
+    with pytest.raises(ValueError):
+        await fm.edit_message('notafact', 'nope', 'Shatt', 'This is not a fact')
+
+
+@pytest.mark.asyncio
+async def test_edit_message(test_fm_fx):
+    """
+    Test edit_message functionally works.
+    """
+    fm = test_fm_fx
+    await fm.edit_message('test', 'en', 'Shatt', 'This fact has been edited during testing.')
+
+    result = await fm.find('test', 'en')
+
+    assert result.message == 'This fact has been edited during testing.'
+
+
+@pytest.mark.asyncio
+async def test_edit_message_failure(test_fm_fx):
+    """
+    Verify an error is raised if an edit on a non-existent fact is attempted.
+    """
+    fm = test_fm_fx
+
+    with pytest.raises(ValueError):
+        await fm.edit_message('nopeavi', 'nope', 'Kovacs', 'This is not a fact')
+
+
+@pytest.mark.asyncio
+async def test_fact_exists(test_fm_fx):
+    """
+    Verify a fact looked up with the exists method returns true.
+    """
+    fm = test_fm_fx
+    assert bool(await fm.exists('stats', 'en'))
+
+
+@pytest.mark.asyncio
+async def test_fact_exists_no(test_fm_fx):
+    """
+    Verify a fact looked up with the exist method returns false if not found.
+    """
+    fm = test_fm_fx
+    assert not bool(await fm.exists('asdf', 'asdfa'))
+
+
+@pytest.mark.asyncio
+async def test_facthistory_return(test_fm_fx):
+    """
+    Verify fact history returns log entries.
+    """
+    fm = test_fm_fx
+
+    # Add some log entries.
+    await fm.log('stats', 'en', 'Shatt', 'Added')
+    await fm.log('stats', 'en', 'Shatt', 'Deleted')
+    await fm.log('stats', 'en', 'Shatt', 'Added')
+    await fm.log('stats', 'en', 'Shatt', 'Marked for Delete')
+
+    history = await fm.facthistory('stats', 'en')
+    assert history is not None
+
+
+@pytest.mark.asyncio
+async def test_facthistory_invalid_query(test_fm_fx):
+    """
+    Verify fact history raises no errors on an invalid fact lookup.
+    """
+    fm = test_fm_fx
+
+    history = await fm.facthistory('s205gw', 'sdjn3nw')
```

Comment:
No assertions?

Reply:
```Verify fact history raises no errors on an invalid fact lookup.```

We are verifying that no error is thrown when looking up an invalid fact, so no assertion is necessary.

Reply:
Fair. Withdrawn.

grumble *GitHub won't let me mark my own comments as resolved...* grumble